### PR TITLE
added the stop button

### PR DESCRIPTION
**Pull request by Sanjay8602**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -58,20 +58,29 @@ def main():
     if topic:
         model1_messages = []
         model2_messages = []
+        stop = False
         for _i in range(st.session_state.exchange):
```
Comment (ogbanugot) [inline app.py:68]: (Line 62) you're still using the `exchange` state in the for loop. I suggest you use a while loop on the stop condition:
```
while not stop:
     ....
```

File: app.py [hunk 2]
```diff
@@ -58,20 +58,29 @@ def main():
...
             with st.chat_message(name="model1", avatar="robot_icon_green.png"):
+                if stop:
```
Comment (ogbanugot) [inline app.py]: this should come immediately under the while loop:
```
while not stop:
     if stop:
          break
```

File: app.py [hunk 3]
```diff
@@ -58,20 +58,29 @@ def main():
...
+                    break
                 if len(model1_messages) == 0:
                     stream = generate_response(model1, topic, "for", [{"role": "user", "content": "start debate."}])
                 else:
                     model1_messages.append({"role": "user", "content": model2_response})
                     stream = generate_response(model1, topic, "for", model1_messages)
                 model1_response = st.write_stream(stream)
+                if st.session_state.stop_button.button("Stop Interaction"):
```
Comment (ogbanugot) [inline app.py]: move this up as well:
```
while not stop:
     if stop:
          break
     if st.session_state.stop_button.button("Stop Interaction"):
           stop = True
```
OR You can just do:
```
while True:
     if st.session_state.stop_button.button("Stop Interaction"):
           break
```
that's even better

File: app.py [hunk 4]
```diff
@@ -58,20 +58,29 @@ def main():
...
+                    stop = True
             model1_messages.append({"role": "assistant", "content": model1_response})
 
             with st.chat_message(name="model2", avatar="robot_icon_yellow.png"):
+                if stop:
```
Comment (ogbanugot) [inline app.py]: remove this duplicate

File: app.py [hunk 5]
```diff
@@ -58,20 +58,29 @@ def main():
...
+                    break
                 model2_messages.append({"role": "user", "content": model1_response})
                 stream = generate_response(model2, topic, "against", model2_messages)
                 model2_response = st.write_stream(stream)
+                if st.session_state.stop_button.button("Stop Interaction"):
```
Comment (ogbanugot) [inline app.py]: remove this duplicate

**Review by ogbanugot — CHANGES_REQUESTED** [review]

**Review by Sanjay8602 — COMMENTED** [review]
updated the stop button

**Pull request closed without merging**